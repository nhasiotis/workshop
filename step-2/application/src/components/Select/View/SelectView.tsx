import React from "react";

export interface SelectProps {
    items: string[];
    id: string;
}

const SelectView: React.FC<SelectProps> = ({ items, id }): JSX.Element => {
    return (
        <select data-id={id}>
            {items.map((item, index) => (
                <option key={index}>{item}</option>
            ))}
        </select>
    );
};

export default SelectView;
