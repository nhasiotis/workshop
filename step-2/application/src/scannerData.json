{
    "scanners": [
        "scanner 1",
        "scanner 2",
        "scanner 3",
        "scanner 4",
        "scanner 5"
    ]
}
