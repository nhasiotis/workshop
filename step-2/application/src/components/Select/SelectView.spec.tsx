import { shallow } from "enzyme";
import React from "react";

import * as scannerData from "../../scannerData.json";
import SelectView from "./SelectView";

describe("Given SelectView", () => {
  describe("when rendering", () => {
    const items: string[] = scannerData.scanners;
    const id = "testId";

    it("should have select component with id and empty list of options matching snapshot", () => {
      const component = shallow(<SelectView items={[]} id={id} />);
      expect(component).toMatchSnapshot();
    });

    it("should have select component with id and list with 5 options matching snapshot", () => {
      const component = shallow(<SelectView items={items} id={id} />);
      expect(component).toMatchSnapshot();
    });
  });
});
