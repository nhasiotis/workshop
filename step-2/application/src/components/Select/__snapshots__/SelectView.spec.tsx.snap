// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given SelectView when rendering should have select component with id and empty list of options matching snapshot 1`] = `
<select
  data-id="testId"
/>
`;

exports[`Given SelectView when rendering should have select component with id and list with 5 options matching snapshot 1`] = `
<select
  data-id="testId"
>
  <option
    key="0"
  >
    scanner 1
  </option>
  <option
    key="1"
  >
    scanner 2
  </option>
  <option
    key="2"
  >
    scanner 3
  </option>
  <option
    key="3"
  >
    scanner 4
  </option>
  <option
    key="4"
  >
    scanner 5
  </option>
</select>
`;
